The use of PASSWORD_HASHERS should be documented more explicitly

When trying to set up I could not get authorization working for quite a while. Then I noticed that this library uses PASSWORD_HASHERS for hashing. For a general case it would work, but people running custom setups like me, may run into issues. In my case I had BCryptPasswordHasher as the top choice and the authorization fails with it. PBKDF2PasswordHasher works fine. Haven't really tested the others. So may I suggest a small section in README about this, along with a list of supported hashers.

@florimondmanca 

> I think it's a good idea to allow to pass a hasher object directly, but how? I can hardly see this package not use the settings infrastructure (similarly to how user passwords gets the hasher from settings).

I'm trying, but having issues.

```sh
                raise ValueError("Unknown password hashing algorithm '%s'. "
                                 "Did you specify it in the PASSWORD_HASHERS "
>                                "setting?" % algorithm)
E               ValueError: Unknown password hashing algorithm 'django.contrib.auth.hashers.BCryptSHA256PasswordHasher'. Did you specify it in the PASSWORD_HASHERS setting?
```

This is with:

```diff
-PASSWORD_HASHERS = ['django.contrib.auth.hashers.BCryptPasswordHasher']
+PASSWORD_HASHERS = [
+    'django.contrib.auth.hashers.BCryptPasswordHasher',
+    'django.contrib.auth.hashers.BCryptSHA256PasswordHasher',  # used by api keys
+]
```

(Note: I have djangorestframework-api-key vendorized / copied into my project)

```diff
+API_KEY_HASHER: str = 'django.contrib.auth.hashers.BCryptSHA256PasswordHasher'
+
+
 def generate_key() -> Tuple[str, str]:
+    from django.conf import settings
+
+    print(settings.PASSWORD_HASHERS)
     prefix = get_random_string(_PREFIX_LENGTH)
     secret_key = get_random_string(_SECRET_KEY_LENGTH)

     key = prefix + "." + secret_key  # for the client
-    key_id = prefix + "." + make_password(key)  # for database storage
+    key_id = (
+        prefix + "." + make_password(key, hasher=API_KEY_HASHER)
+    )  # for database storage

     return key, key_id


 def check_key(key: str, hashed_key: str) -> bool:
-    return check_password(key, hashed_key)
+    return check_password(key, hashed_key, preferred=API_KEY_HASHER)
```

For reference: https://docs.djangoproject.com/en/2.2/_modules/django/contrib/auth/hashers/#make_password

--- a/rest_framework_api_key/models.py
+++ b/rest_framework_api_key/models.py
@@ -20,10 +20,7 @@ class APIKeyManager(models.Manager):
         return obj, generated_key
 
     def is_valid(self, key: str) -> bool:
-        try:
-            prefix, _, _ = key.partition(".")
-        except ValueError:
-            return False
+        prefix, _, _ = key.partition(".")
 
         try:
             obj = self.get(id__startswith=prefix, revoked=False)

--- a/rest_framework_api_key/models.py
+++ b/rest_framework_api_key/models.py
@@ -3,7 +3,7 @@ from typing import Tuple
 from django.core.exceptions import ValidationError
 from django.db import models
 
-from ._helpers import generate_key, check_key
+from ._helpers import check_key, generate_key
 
 
 class APIKeyManager(models.Manager):
@@ -21,7 +21,7 @@ class APIKeyManager(models.Manager):
 
     def is_valid(self, key: str) -> bool:
         try:
-            prefix, _ = key.split(".")
+            prefix, _, _ = key.partition(".")
         except ValueError:
             return False
 
@@ -56,14 +56,14 @@ class APIKey(models.Model):
         self._initial_revoked = self.revoked
 
     def prefix(self) -> str:
-        return self.pk.split(".")[0]
+        return self.pk.partition(".")[0]
 
     # Ensure compatibility with Django admin.
     prefix.short_description = "Prefix"
     prefix = property(prefix)
 
     def is_valid(self, key: str) -> bool:
-        _, hashed_key = self.pk.split(".")
+        _, _, hashed_key = self.pk.partition(".")
         return check_key(key, hashed_key)
 
     def clean(self):

--- a/tests/test_api_key_model.py
+++ b/tests/test_api_key_model.py
@@ -10,7 +10,7 @@ pytestmark = pytest.mark.django_db
 
 def test_key_generation():
     api_key, generated_key = APIKey.objects.create_key(name="test")
-    prefix, hashed_key = api_key.id.split(".")
+    prefix, _, hashed_key = api_key.id.partition(".")
 
     assert prefix and hashed_key
 

--- a/tests/test_has_api_key.py
+++ b/tests/test_has_api_key.py
@@ -25,7 +25,7 @@ def test_if_no_api_key_then_permission_denied(create_request, view):
 
 
 def _scramble_prefix(key: str) -> str:
-    prefix, secret_key = key.split(".")
+    prefix, _, secret_key = key.partition(".")
     truncated_prefix = prefix[:-1]
     return truncated_prefix + "." + secret_key
 

